### DISCUSSION TITLE
strip=sometimes should strip in both "-c fastbuild" and "-c opt"

### DISCUSSION THREAD

**Discussion by aran** [bazelbuild/bazel#24149] [General]
The default value of `--strip` is documented as follows:

```
[--strip](https://bazel.build/reference/command-line-reference#flag--strip)=<always, sometimes or never> default: "sometimes"
Specifies whether to strip binaries and shared libraries (using "-Wl,--strip-debug"). The default value of 'sometimes' means strip iff --compilation_mode=fastbuild.
```

This results in the following behavior:

`bazel build :target          # fastbuild, stripped`
`bazel build :target -c opt   # opt, not stripped(!)`

It is surprising that switching to "-c opt" could result in the inclusion of debug symbols and the resulting artifact being larger than the default. My expectation would be that "-c opt" would strip debug symbols by default.

This documentation was explicitly used as a spec in rules_go: https://github.com/bazel-contrib/rules_go/issues/3917, resulting in binary size surprisingly growing when "-c opt" is enabled.

It would be more intuitive if the spec was:

```
[--strip](https://bazel.build/reference/command-line-reference#flag--strip)=<always, sometimes or never> default: "sometimes"
Specifies whether to strip binaries and shared libraries (using "-Wl,--strip-debug"). The default value of 'sometimes' means strip iff --compilation_mode=fastbuild or --compilation_mode=opt.
```

**Comment by keith** [comment]
I imagine the intent here is that with opt binaries you might want to collect debug info from the binary for debugging crashes later, and then strip it yourself. Note that cc_binary has a `.stripped` implicit target you can use that is always stripped https://bazel.build/reference/be/c-cpp#cc_binary

**Reply by aran** [reply]
Interesting. I think it's a strange default. But I don't have data about other developer expectations.
